beetsplug/replaygain.py

```diff
@@ -308,42 +307,47 @@ def compute_album_gain(self, task):
         target_level_lufs = db_to_lufs(task.target_level)
 
         # analyse tracks
-        # list of track Gain objects
-        track_gains = []
-        # maximum peak
-        album_peak = 0
-        # sum of BS.1770 gating block powers
-        sum_powers = 0
-        # total number of BS.1770 gating blocks
-        n_blocks = 0
-
-        for item in task.items:
-            track_gain, track_n_blocks = self._analyse_item(
-                item, task.target_level, task.peak_method
+        # Gives a list of tuples (track_gain, track_n_blocks)
+        track_results = [
+            self._analyse_item(
+                item,
+                task.target_level,
+                task.peak_method,
+                count_blocks=True,
             )
-            track_gains.append(track_gain)
+            for item in task.items
+        ]
+
+        # list of track Gain objects
+        track_gains = [tg for tg, _nb in track_results]
 
-            # album peak is maximum track peak
-            album_peak = max(album_peak, track_gain.peak)
+        # Album peak is maximum track peak
+        album_peak = max(tg.peak for tg in track_gains)
 
-            # prepare album_gain calculation
-            # total number of blocks is sum of track blocks
-            n_blocks += track_n_blocks
+        # Total number of BS.1770 gating blocks
+        n_blocks = sum(nb for _tg, nb in track_results)
```

Comment:
It's very satisfying that we can now use `max` and `sum` here, as ~God~ Guido intended.